src/test/java/com/algorand/algosdk/logic/TestLogic.java

```diff
@@ -264,4 +264,55 @@ public void testCheckProgramTealV3() throws Exception {
             assertThat(valid).isTrue();
         }
     }
+
+    @Test
+    public void testCheckProgramTealV4() throws Exception {
+        assertThat(getEvalMaxVersion()).isGreaterThanOrEqualTo(4);
+        assertThat(getLogicSigVersion()).isGreaterThanOrEqualTo(3);
```

Comment:
What's the difference between EvalMaxVersion and LogicSigVersion? Is there a reason we kept LogicSigVersion at 3, and is the expectation that we'll update this to 4 at some point?

@jannotti may be able to answer this.

Reply:
```
	return &LanguageSpec{
		EvalMaxVersion:  logic.LogicVersion,
		LogicSigVersion: config.Consensus[protocol.ConsensusCurrentVersion].LogicSigVersion,
		Ops:             records,
	}
```
logic.LogicVersion is a "normal" Go variable, so it might be incremented was we work on, say, Teal 4, and even we release that, it will mean `goal` can compile v4,  but the network still won't evaluate it.  The consensus variable determines what the network will do.

So this test is running with a v4 compatible assembler, but a network that only takes v3 programs.  Is that the intent?

Reply:
Really I think it's just expressing that it wants the langspec.json file to have those values.  I don't know why.  I suspect it should not be testing the consensus version at all, as that is the consensus version when the langspec was written.  I don't understand why that's ever relevant to anyone. (Though I bet it is, I just can't think why right now!)

Reply:
```suggestion
        // TODO: check LogicSigVersion once v4 is supported by consensus
        // assertThat(getLogicSigVersion()).isGreaterThanOrEqualTo(4);
```
Thanks, that explanation makes sense.

It seems misleading to check that `LogicSigVersion` is greater to or equal to 3 in this test. I'd prefer one of:
1. Change the check to greater than or equal to 4, but comment it out for now (as in my suggested change).
2. Don't check `LogicSigVersion` at all, since it's dependent on consensus and the SDK doesn't need to care about that.

Reply:
You learn something new everyday - I didn't know you could do suggested changes! That's pretty nice. (though in this case I lean toward #2, unless we understand why it's worth testing the consensus value)